hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/TestHoodieMergeOnReadTable.java

```diff
@@ -271,6 +282,84 @@ public void testLogFileCountsAfterCompaction(boolean populateMetaFields) throws
     }
   }
 
+  @Test
+  public void testOutOfSyncView() throws Exception {
+    // insert 100 records
+    HoodieWriteConfig.Builder cfgBuilder1 = getConfigBuilder(false, false, IndexType.BLOOM,
+        1024 * 1024 * 1024L, HoodieClusteringConfig.newBuilder().build(), 3)
+        .withWriteConcurrencyMode(WriteConcurrencyMode.OPTIMISTIC_CONCURRENCY_CONTROL)
+        .withMetadataConfig(HoodieMetadataConfig.newBuilder().enable(false).build());
+    addConfigsForPopulateMetaFields(cfgBuilder1, true);
+    Properties properties = new Properties();
+    properties.setProperty(FILESYSTEM_LOCK_PATH_PROP_KEY, basePath() + "/.hoodie/.locks");
+    properties.setProperty(LOCK_PROVIDER_CLASS_NAME.key(), "org.apache.hudi.client.transaction.FileSystemBasedLockProviderTestClass");
+    cfgBuilder1.withProperties(properties);
+    HoodieWriteConfig config1 = cfgBuilder1.build();
+
+    HoodieWriteConfig.Builder cfgBuilder2 = getConfigBuilder(false, false, IndexType.BLOOM,
+        1024 * 1024 * 1024L, HoodieClusteringConfig.newBuilder().build(), 1)
+        .withWriteConcurrencyMode(WriteConcurrencyMode.OPTIMISTIC_CONCURRENCY_CONTROL)
+        .withMetadataConfig(HoodieMetadataConfig.newBuilder().enable(false).build());
+    addConfigsForPopulateMetaFields(cfgBuilder1, true);
```

Comment:
do we need this? simpler the test , relying on defaults as much as possible the better